File: 2.8.noaa/Station.py
from __future__ import unicode_literals
from hdfs3 import HDFileSystem

test_host = 'localhost'
test_port = 9000

class Station:
    
    def __init__(self, USAF, NAME, FIPS, ELEV):
        self.USAF = USAF
        self.NAME = NAME
        self.FIPS = FIPS
        self.ELEV = ELEV
    
    def display(self):
        print("USAF:", self.USAF, " NAME:", self.NAME, " FIPS:", self.FIPS, " ELEV:", self.ELEV)

if __name__ == '__main__':
    
    file = '/test/isd-history.txt'
    station = []
    n = 0
    
    # connect to HDFS and read the file
    hdfs_client = HDFileSystem(host=test_host, port=test_port)

    with hdfs_client.open(file, 'rb') as f:
        line = f.readline()
        while line:
            n += 1
            if n > 22 :
                USAF = line[0:6]
                NAME = line[13:42]
                FIPS = line[43:45]
                ELEV = line[74:81]
                print("I'm converting the number " + str(n) + " station ...")
                station.append(Station(USAF, NAME, FIPS, ELEV))
            line = f.readline()
    
    hdfs_client.disconnect()
    
    # display the station info
    n -= 22
    print("There are", n, "stations :")
    for i in range(n):
        print(i, end=' ')
        station[i].display()
    
    print("-" * 20 + "END" + "-" * 20)

File: 2.8.noaa/Station.java
package Station;

import java.io.*;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.fs.FSDataInputStream;
import org.apache.hadoop.fs.FileSystem;
import org.apache.hadoop.fs.Path;

public class Station {

    public static void main() throws IOException{

        String localSrc = "/test/isd-history.txt";
        Path path d = new Path(localSrc);
        
        //Open the file
        Configuration conf = new Configuration();
        FileSystem fs = FileSystem.get(conf);
        FSDataInputStream in = fs.open(path);

        try {

            InputStreamReader isr = new InputStreamReader(in);
            BufferedReader br = new BufferedReader(isr);
            
            int n = 0;
            br.readline();
            
            while (br != null) {
                
                n = n + 1
                if (n > 22) {
                    
                    // Process of the current line
                    System.out.println(br.readLine().substring(0, 7) + " " + br.readLine().substring(13, 42) + " " + br.readLine().substring(43, 45) + " " + br.readLine().substring(74, 81));
                    // go to the next line
                }
                br.readLine();
            }
        }

        finally {
            //close the file
            in.close();
            fs.close();
        }
    }
}

